server/main.lua

```diff
@@ -671,12 +671,40 @@ local function SaveOwnedVehicleItems(plate, items)
 	Trunks[plate].isOpen = false
 end
 
-local function AddToTrunk(plate, slot, otherslot, itemName, amount, info, created)
+local function FindEmptyTrunkSlot(items, vehicleName, vehicleClass)
+	local slots = Config.VehicleInventories.vehicles[vehicleName] or Config.VehicleInventories.classes[vehicleClass] or Config.VehicleInventories.default
+    for i = 1, slots.slots do
+        if items[i] == nil then
+            return i
+        end
+    end
+    return nil
+end
+
+local function AddToTrunk(plate, slot, otherslot, itemName, amount, info, created, vehicleName, vehicleClass)
```

Comment:
While at it, splitting this method into smaller methods for readability would be nice.